Sources/App/Commands/TriggerBuilds.swift

```diff
@@ -184,68 +173,76 @@ func triggerBuilds(on database: Database,
 ///   - logger: `Logger` used for logging
 ///   - packages: list of `Package.Id`s to trigger
 ///   - force: do not check pipeline capacity and ignore downscaling
-/// - Returns: `EventLoopFuture<Void>` future
 func triggerBuilds(on database: Database,
                    client: Client,
                    logger: Logger,
                    packages: [Package.Id],
-                   force: Bool = false) -> EventLoopFuture<Void> {
+                   force: Bool = false) async throws {
     guard Current.allowBuildTriggers() else {
         logger.info("Build trigger override switch OFF - no builds are being triggered")
-        return database.eventLoop.future()
+        return
     }
 
     guard !force else {
-        return packages.map {
-            findMissingBuilds(database, packageId: $0).flatMap {
-                triggerBuildsUnchecked(on: database, client: client, logger: logger, triggers: $0) }
+        return await withThrowingTaskGroup(of: Void.self) { group in
+            for package in packages {
+                group.addTask {
+                    let triggerInfo = try await findMissingBuilds(database, packageId: package)
+                    try await triggerBuildsUnchecked(on: database, client: client, logger: logger, triggers: triggerInfo)
+                }
+            }
```

Comment:
TODO: double check if this is the correct conversion. We might be making the triggers concurrent when they're not in ELF-land.

Reply:
https://gist.github.com/finestructure/91f7d755c7d966751fd1a23e64b03013